src/pathfinder/water_regions.cpp

```diff
@@ -267,8 +268,11 @@ WaterRegionPatchDesc GetWaterRegionPatchInfo(TileIndex tile)
 void InvalidateWaterRegion(TileIndex tile)
 {
 	const int index = GetWaterRegionIndex(tile);
-	if (index > static_cast<int>(_water_regions.size())) return;
+	if (index >= static_cast<int>(_water_regions.size())) return;
 	_water_regions[index].Invalidate();
+
+	// TODO KB remove
+	Debug(map, 2, "Invalidated water region ({},{})", GetWaterRegionX(tile), GetWaterRegionY(tile));
```

Comment:
TODO, but not a draft?

Generally speaking, a PR can be a draft (with TODOs) or be ready-for-review (without TODOs). Makes it easier for us how to look at code :)

Reply:
My bad, should be fixed now